aezeed/cipherseed.go

```diff
@@ -128,6 +128,33 @@ var (
 	BitcoinGenesisDate = time.Unix(1231006505, 0)
 )
 
+// SeedOptions is a type that holds options that configure the generation of a
+// new cipher seed.
+type SeedOptions struct {
+	// randomnessSource is the source of randomness that is used to generate
+	// the salt that is used for encrypting the seed.
+	randomnessSource io.Reader
+}
+
+// DefaultOptions returns the default seed options.
+func DefaultOptions() *SeedOptions {
+	return &SeedOptions{
+		randomnessSource: rand.Reader,
```

Comment:
Does this actually compile? AFAICT there's no struct/method that implements the `io.Reader` API. While working on a recent change, I ended up defining a new type that mirrors the `Read` function for the `rand` package.

Reply:
Link to my prior work around: https://github.com/btcsuite/btcd/blob/master/btcec/schnorr/musig2/nonces.go#L94-L142

Reply:
Ok nvm, turns out it is a real thing after all: https://pkg.go.dev/crypto/rand#Read 

Might be new?

Reply:
I went through the official docs and this seems to have been there since go 1.1...